src/wallet/wallet.cpp

```diff
@@ -2839,6 +2839,9 @@ std::shared_ptr<CWallet> CWallet::Create(WalletContext& context, const std::stri
 
         walletInstance->InitWalletFlags(wallet_creation_flags);
 
+        // Set USES_KEYMAN to prevent auto upgrade from trying to auto upgrade (previously) blank wallets that had imported things
+        walletInstance->SetWalletFlag(WALLET_FLAG_USES_KEYMAN);
```

Comment:
eb75f8d58ec335e013178d5b94d824cc7739139a: maybe put this under the `if (!walletInstance->IsWalletFlagSet(WALLET_FLAG_DESCRIPTORS))` check below?